Theory assessment/Theory_assessment_Software_Engineering.md

```diff
@@ -0,0 +1,150 @@
+### 1. How does Object-oriented Programming differ from Process Oriented Programming? 
+A: Object-oriented programming is divided into the objects, but process-oriented programming is divided into smaller functions. OOP has bottom-up approach and data members can be public/private/default, which can also help hide the data and is more secure. POP has top-to-down approach and the data members are public, also data hiding is impossible. 
+Function overloading in OOP is possible, but not possible in POP. OOP is great for solving big problems, and it focuses on data, POP focuses on functions.
+
+### 2. What's polymorphism in OOP?
+A: Polymorphism in OOP means you can use same function for different purposes, used and reused many times. Example: you can overwrite something that has different outcome in parent class. Has positive influence to code, because can save time and makes easy to debug if needed.
+```
+class Animals:
+    def __init__(self, name, age):
+        self.name = name
+        self.age = age
+        
+    def lives_in_farm(self):
+        return True
+            
+class Dog(Animals):
+    def __init__(self, name, age):
+        super().__init__(name, age)
+        
+    def lives_in_farm(self):
+        return False
+
+dog_Mika=Dog('Mika', 3)
+print(dog_mika.name, dog_mika.age, dog_mika.lives_in_farm())
+```
+### 3. What's inheritance in OOP?
+A: Inheritance in OOP means that we can transfer methods/properties from one class to another. It helps with code reading, reusing and scaling and decreases repetition. Also better and easier to find errors.
+```
+class Animals:
+    def __init__(self, name, age):
+        self.name = name
+        self.age = age
+        
+    def lives_in_farm(self):
+        return True
+            
+class Cow(Animals):
+    def __init__(self, name, age):
+        super().__init__(name, age)       
+          
+cow_alma=Cow('Alma', 12)
+print(cow_alma.name, cow_alma.age, cow_alma.lives_in_farm())
+```
+### 4. If you had to make a program that could vote for the top three funniest people in the office, how would you do that? How would you make it possible to vote on those people?
+- APPROACH: I would firstly make plan, determine needed tools and program architecture. I think in this case it would be great if there would be e-mail link for voting, so everybody could vote one time and they could be identified by e-mail, so we could keep track how has voted and send reminders. The program has multiple choice answer questions, where you can only vote for one person. If there are a lot of different departments, then I would make extra choice, to see who are in which department. After voting is ended, you can see who are three people who got most votes. When somebody is in a tie, then both names are shown.
+- KEY REQUIREMENTS:
```

Comment:
8/10 Missing one important point in the key requirements. i.e. how do you measure the 'funny' is there any attribute you want to display to the voter? i.e. the hobby, photos, or jokes posted by the employee.